import "ydb/core/protos/console_base.proto";
import "ydb/core/protos/console_tenant.proto";
import "ydb/core/protos/console_config.proto";

package NKikimrConsole;
option java_package = "ru.yandex.kikimr.proto";

message TConfig {
    optional TTenantsConfig TenantsConfig = 1;
    optional TConfigsConfig ConfigsConfig = 2;
}

message TGetConfigRequest {
}

message TGetConfigResponse {
    optional TConfig Config = 1;
}

message TSetConfigRequest {
    optional TConfig Config = 1;
    optional TConfigItem.EMergeStrategy Merge = 2 [default = OVERWRITE];
}

message TSetConfigResponse {
    optional TStatus Status = 1;
}

message TVolatileConfig {
    optional uint64 Id = 1;
    optional string Config = 2;
}

message TYamlConfigChange {
    optional string OldYamlConfig = 1;
    optional string NewYamlConfig = 2;
    repeated TVolatileConfig OldVolatileYamlConfigs = 3;
    repeated TVolatileConfig NewVolatileYamlConfigs = 4;
}

message TLogRecordData {
    // Old configs
    optional TConfigureRequest Action = 1;
    repeated TConfigureResponse.TAffectedConfig AffectedConfigs = 2;
    repeated uint32 AffectedKinds = 3;
    // New configs
    optional TYamlConfigChange YamlConfigChange = 4;
}

message TLogRecord {
    optional uint64 Id = 1;
    optional uint64 Timestamp = 2;
    optional string User = 3;
    optional TLogRecordData Data = 4;
}

message TLogFilter {
    optional uint64 FromId = 1;
    optional uint64 FromTimestamp = 2;
    optional uint32 Limit = 3 [default = 100];
    repeated uint32 AffectedKinds = 4;
    repeated string Users = 5;
    repeated string ExcludeUsers = 6;
    optional bool Reverse = 7;
}

message TGetLogTailRequest {
    optional TLogFilter LogFilter = 1;
}

message TGetLogTailResponse {
    optional TStatus Status = 1;
    repeated TLogRecord LogRecords = 4;
}
